### DISCUSSION TITLE
What is the default sample size with alloc event ?

### DISCUSSION THREAD

**Discussion by ahamlat** [async-profiler/async-profiler#912] [Q&A] [answered]
Hi,
Sorry if it already shared somewhere, but I couldn't find the answer, I'm looking for the default size when sampling with alloc event, like this :
`sudo bin/asprof -d 60 -e cpu,alloc,lock -f /tmp/CPU-profiling-JFR-output.jfr pid`

**Comment by apangin** [comment, marked as answer]
You should not assume there is a fixed default sampling interval for allocation profiling. In fact, in JDK versions prior to JDK 11, the interval is floating and depends on the current TLAB size. In JDK 11+, the default interval is ~512 KiB, but as an implementation detail, it is intentionally left unspecified and may change in future versions of the profiler without a notice.

Moreover, `-e cpu,alloc,lock` syntax is deprecated. The recommended syntax for multi-event profiling is
```
-e cpu --alloc 2m --lock 1ms
```
which explicitly specifies the allocation profiling interval and the contended lock threshold.

**Reply by ahamlat** [reply]
Thanks @apangin
